qt/main.cpp

```diff
@@ -215,20 +214,19 @@ int main(int argc, char * argv[])
     fmt.setSwapInterval(1);
     fmt.setDepthBufferSize(16);
 
-    // This is a workaround for some systems,
-    // including MacOs and older Linux distros, which rely on X.org and Mesa.
-    // Where somehow the driver itself doesn't
-    // make all the otherwise supported GLSL versions available by default
-    // and such requests are somehow disregarded at later stages of execution.
-    // This setting here will be potentially overwritten and overruled anyway,
-    // and only needed to ensure that we have the needed GLSL compiler available
-    // later when we need it.
-    if (app.platformName() == QString("xcb") ||
-        app.platformName() == QString("cocoa"))
+    if (app.platformName() == QString("wayland"))
+    {
+      // TODO: Check if there is a way to use OpenGL 3.2 on wayland platform.
+      fmt.setProfile(QSurfaceFormat::CompatibilityProfile);
+      fmt.setVersion(2, 0);
+    }
+    else
     {
+      // TODO: Avoid copy-pasting the same version/profile in map_widget.cpp and test_main_loop.cpp
```

Comment:
Since you have mentioned the code duplication with `test_main_loop.cpp` shouldn't we at least synch in this commit? In `test_main_loop.cpp` it is still `2.1` which is tested, while in the production code now we only have `2.0`.